backend/app/consents.py

```diff
@@ -125,11 +126,14 @@ async def get_consent_verification_url(dsi: str, sub: str) -> Optional[str]:
     :param sub: Subject of consent
     :return: URL to verify a consent in Consent Portal
     """
-    url = f"{conf.CONSENT_PORTAL_URL}/Consent/RequestConsents"
     cr_token = create_consent_request_token(sub)
     headers = {"X-Consent-Request-Token": cr_token}
     payload = {"consentRequests": [{"dataSource": dsi, "required": True}]}
     async with httpx.AsyncClient() as client:
+        dataspace_configuration = await get_dataspace_configuration()
+        consent_provider = dataspace_configuration["consent_providers"][0]
+        consent_provide_base_url = consent_provider["base_url"]
+        url = f"{consent_provide_base_url}/Consent/RequestConsents"
```

Comment:
I guess I'd in one step extracted the URL from it like this:
```suggestion
        consent_portal_url = dataspace_configuration["consent_providers"][0]["base_url"]
        url = f"{consent_portal_url}/Consent/RequestConsents"
```
but really no need to change it.

Reply:
And I think I'd also left it where the original place was for defining the URL instead of inside the `async with` block.